Source/JavaScriptCore/wasm/WasmMemory.cpp

```diff
@@ -494,14 +494,15 @@ bool Memory::addressIsInGrowableOrFastMemory(void* address)
 
 Expected<PageCount, Memory::GrowFailReason> Memory::growShared(PageCount delta)
 {
-#if CPU(ARM)
-    // Shared memory requires signaling memory which is not available on ARMv7
+#if !ENABLE(WEBASSEMBLY_SIGNALING_MEMORY)
+    // Shared memory requires signaling memory which is not available on ARMv7 or others
     // yet. In order to get more of the test suite to run, we can still use
     // a shared mmeory by using bounds checking, but we cannot grow it safely
     // in case it's used by multiple threads. Once the signal handler are
     // available, this can be relaxed.
-    RELEASE_ASSERT_NOT_REACHED();
+    return makeUnexpected(GrowFailReason::GrowSharedUnavailable);
```

Comment:
Would this cause a compiler warning / error due to the rest of this function not being reachable?

Reply:
I don't know since I don't have ARMv7, but so, for now, lets put the rest in #else.